updatersync/src/main/java/nl/openweb/hippo/groovy/watch/GroovyFilesWatcher.java

```diff
@@ -82,11 +81,13 @@ private FileSystemObserver observeFileSystem(final Path projectBaseDir) {
         }
 
         List<Path> groovyFilesDirectories = WatchFilesUtils.getGroovyFilesDirectories(projectBaseDir, config);
-        log.debug("Observe {} paths: {}", groovyFilesDirectories.size(), groovyFilesDirectories.stream().map(Path::toString)
-                .collect(Collectors.joining(", ")));
+        if(log.isDebugEnabled()) {
```

Comment:
Are you getting so much NPE when not doing this?

Reply:
I don't get the question

Reply:
log.isDebugEnabled is also inside log.debug(). I see this line of code repeated many time.
Resulting in the question are you getting NPE's when you would remove them?

Reply:
In this case it makes a difference, since the logic to get to the parameters isn't executed this way. If relying on the internal logic the computing has already happened.

Reply:
Fair enough